src/photon/photon_scheduler.c

```diff
@@ -89,18 +89,12 @@ void assign_task_to_worker(local_scheduler_state *state,
   write_message(w->sock, EXECUTE_TASK, task_spec_size(spec), (uint8_t *) spec);
   /* Update the global task table. */
   if (state->db != NULL) {
-    retry_info retry;
-    memset(&retry, 0, sizeof(retry));
-    retry.num_retries = 0;
-    retry.timeout = 100;
-    retry.fail_callback = NULL;
     task *task =
         alloc_task(spec, TASK_STATUS_RUNNING, get_db_client_id(state->db));
-    if (from_global_scheduler) {
-      task_table_update(state->db, task, (retry_info *) &retry, NULL, NULL);
-    } else {
-      task_table_add_task(state->db, task, (retry_info *) &retry, NULL, NULL);
-    }
+    task_table_update(state->db, task, (retry_info *) &photon_retry, NULL,
+                      NULL);
+    /* Record which task this worker is executing. */
+    w->task_in_progress = copy_task(task);
```

Comment:
Oh I see, in `handle_worker_available`, we are passing it into `task_table_update`, which takes ownership? Hmm... that feels a bit suspect. Maybe we should just pass a copy into `task_table_update` and free it in `handle_worker_available`?

If not, it should at least be documented what is going on.